packages/auth/src/providers/email-password.ts

```
@@ -16,6 +16,14 @@ class EmailPasswordProvider extends AbstractAuthModuleProvider {
     this.authUserSerivce_ = authUserService
   }
 
+  private getHashConfig(scope: string) {
+    const scopeConfig = this.scopes_[scope].hashConfig as
```

This is a lot of trust on the chaining 😄 

suggestion: throw error or silently fail when scope is not found

---

we do that before going to the provider in the authentication or validateCallback module methods when we verify the provider here: 
```
 protected getRegisteredAuthenticationProvider(
    provider: string,
    { authScope }: AuthenticationInput
  ): AbstractAuthModuleProvider {
    let containerProvider: AbstractAuthModuleProvider
    try {
      containerProvider = this.__container__[`auth_provider_${provider}`]
    } catch (error) {
      ...
    }

    containerProvider.validateScope(authScope)

    return containerProvider
  }
```

we call the abstract class method: 
```
public validateScope(scope) {
    if (!this.scopes_[scope]) {
      throw new MedusaError(
        MedusaError.Types.INVALID_ARGUMENT,
        `Scope "${scope}" is not valid for provider ${this.provider}`
      )
    }
  }
```